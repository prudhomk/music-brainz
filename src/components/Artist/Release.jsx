import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';
import { getArtWork } from '../../services/musicApi';

export default function Release({ id, title, art, date }) {
  const [image, setImage] = useState('');
  
  
  useEffect(async () => {
    const art = await getArtWork(id);
    console.log(art);
    await setImage(art);
 
  }, []);

  return (
    <>  
      <img src={image} alt={title} />
      <h1>{title}</h1>
      <p>{date}</p>
            
    </>
  );
}

Release.propTypes = {
  id: PropTypes.string.isRequried,
  title: PropTypes.string.isRequired,
  date: PropTypes.string.isRequired,
  art: PropTypes.string.isRequired
};
