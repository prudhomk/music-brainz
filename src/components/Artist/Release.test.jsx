import React from 'react';
import { render } from '@testing-library/react';
import { act } from 'react-dom/test-utils';
import Release from './Release';


describe('Release component snapshot', () => {

  it('render Release', () => {
    const release = {
      id: '1',
      title: 'title',
      date: 'date',
      art: 'image',
    };

    act(() => {
      const { asFragment } = render(
        <Release {...release} />
      );
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
