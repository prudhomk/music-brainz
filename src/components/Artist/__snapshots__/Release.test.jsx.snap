// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Release component snapshot render Release 1`] = `
<DocumentFragment>
  <img
    alt="title"
    src=""
  />
  <h1>
    title
  </h1>
  <p>
    date
  </p>
</DocumentFragment>
`;
